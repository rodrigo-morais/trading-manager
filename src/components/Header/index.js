import React from 'react'
import styled from 'styled-components'

import DropDown from '../shared/DropDown'


const Container = styled.div`
  display: inline-grid;
  grid-template-columns: 26.66% 26.66% 26.66%;
  justify-content: space-evenly;
  min-width: 100%;
`

const DropDownLeft = styled(DropDown)`
  grid-column-start: 1;
  grid-column-end: 1;
  justify-self: start;
`
const DropDownMiddle = styled(DropDown)`
  grid-column-start: 3;
  grid-column-end: 3;
  justify-self: center;
`

const DropDownRight = styled(DropDown)`
  grid-column-start: 6;
  grid-column-end: 6;
  justify-self: end;
`

const Header = () => (
  <Container>
    <DropDownLeft label="Ativo" options={['WINM20', 'WDON20']} />
    <DropDownMiddle label="Tipo" options={['Pregão', 'Abertura']} />
    <DropDownRight label="Estratégia" options={['Topo / Fundo', 'Fecha Fora / Fecha Dentro', '3x1', 'Fabolous 4']} />
  </Container>
)

export default Header
