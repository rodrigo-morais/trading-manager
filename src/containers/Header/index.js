import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import DropDown from '../../components/shared/DropDown'


const Container = styled.div`
  display: inline-grid;
  grid-template-columns: 18% 18% 30% 30%;
  justify-content: space-evenly;
  min-width: 100%;
`

const getSystem = (systems, callback) => (id) => {
  callback(systems.find((system) => system.id === id))
}

const getStock = (stocks, callback) => (acronym) => {
  callback(stocks.find((stock) => stock.acronym === acronym))
}

const Header = ({ trades, stocks, systems, className, selectStock, selectType, selectSystem, selectStrategy, disabled }) => {
  const systemName = !!trades.system ? trades.system.name : ''
  const stockName = !!trades.stock ? trades.stock.acronym : ''

  if (!trades.stock) {
    selectStock(stocks[0])
    selectSystem(systems.filter((system) => system.stock === stocks[0].acronym)[0])
  }

  return (
    <Container className={className}>
      <DropDown label="Ativo" options={stocks.map(({ acronym }) => ({ id: acronym, text: acronym }))} value={stockName} onChange={getStock(stocks, selectStock)} disabled={disabled} />
      <DropDown label="Tipo" options={['Pregão', 'Abertura']} value={trades.type} onChange={selectType} disabled={disabled} />
      <DropDown label="Sistema" options={systems.map(({ id, name }) => ({ id, text: name }))} value={systemName} onChange={getSystem(systems, selectSystem)} disabled={disabled} />
      <DropDown label="Estratégia" options={['Topo / Fundo', 'Fecha Fora / Fecha Dentro', '3x1', 'Fabolous 4']} value={trades.strategy} onChange={selectStrategy} disabled={disabled} />
    </Container>
  )
}

Header.defaultProps = {
  className: '',
  disabled: true,
}

Header.propTypes = {
  trades: PropTypes.object.isRequired,
  stocks: PropTypes.array.isRequired,
  systems: PropTypes.array.isRequired,
  className: PropTypes.string,
  selectStock: PropTypes.func.isRequired,
  selectType: PropTypes.func.isRequired,
  selectSystem: PropTypes.func.isRequired,
  selectStrategy: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
}

export default Header
