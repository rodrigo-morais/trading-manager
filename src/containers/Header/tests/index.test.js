import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Header from '..'

const systems = [{ id: 1, name: 'name 1' }, { id: 2, name: 'name 2' }]
const trades = { stock: 'WINM20', type: 'Abertura', system: systems[0], strategy: '3x1' }

describe('Header', () => {
  describe('when is enabled', () => {
    describe('Rendering', () => {
      const wrapper = (
        <Header
          trades={trades}
          systems={systems}
          selectStock={() => {}}
          selectType={() => {}}
          selectSystem={() => {}}
          selectStrategy={() => {}}
          disabled={false}
        />
      )

      it('renders according to design', () => {
        const component = renderer.create(wrapper)
        const tree = component.toJSON()

        expect(tree).toMatchSnapshot()
      })

      it('has the first label as "Ativo"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(0).text()).toEqual('Ativo:')
      })

      it('has the second label as "Tipo"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(1).text()).toEqual('Tipo:')
      })

      it('has the third label as "Sistema"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(2).text()).toEqual('Sistema:')
      })

      it('has the fourth label as "Estratégia"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(3).text()).toEqual('Estratégia:')
      })

      it('has 4 selects', () => {
        const component = mount(wrapper)
        expect(component.find('select').length).toEqual(4)
      })
    })
  })

  describe('when is disabled', () => {
    describe('Rendering', () => {
      const wrapper = (
        <Header
          trades={trades}
          systems={systems}
          selectStock={() => {}}
          selectType={() => {}}
          selectSystem={() => {}}
          selectStrategy={() => {}}
          disabled
        />
      )

      it('renders according to design', () => {
        const component = renderer.create(wrapper)
        const tree = component.toJSON()

        expect(tree).toMatchSnapshot()
      })

      it('has the first label as "Ativo"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(0).text()).toEqual('Ativo:')
      })

      it('has the first label value as "WINM20"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(1).text()).toEqual('WINM20')
      })

      it('has the second label as "Tipo"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(2).text()).toEqual('Tipo:')
      })

      it('has the second label value as "Abertura"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(3).text()).toEqual('Abertura')
      })

      it('has the third label as "Sistema"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(4).text()).toEqual('Sistema:')
      })

      it('has the third label value as "name 1"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(5).text()).toEqual('name 1')
      })

      it('has the fourth label as "Estratégia"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(6).text()).toEqual('Estratégia:')
      })

      it('has the fourth label value as "3x1"', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(7).text()).toEqual('3x1')
      })

      it('has 4 labels instead of selects', () => {
        const component = mount(wrapper)
        expect(component.find('label').length).toEqual(8)
      })
    })
  })
})
