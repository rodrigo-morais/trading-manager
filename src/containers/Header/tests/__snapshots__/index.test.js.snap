// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header when is disabled Rendering renders according to design 1`] = `
<div
  className="sc-ifAKCX LlTRb"
>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Ativo"
    >
      Ativo:
    </label>
    <label
      className="sc-htpNat jMFgIO"
      id="AtivoValue"
      name="AtivoValue"
    >
      WDO
    </label>
  </div>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Tipo"
    >
      Tipo:
    </label>
    <label
      className="sc-htpNat jMFgIO"
      id="TipoValue"
      name="TipoValue"
    >
      Abertura
    </label>
  </div>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Sistema"
    >
      Sistema:
    </label>
    <label
      className="sc-htpNat jMFgIO"
      id="SistemaValue"
      name="SistemaValue"
    >
      name 1
    </label>
  </div>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Estratégia"
    >
      Estratégia:
    </label>
    <label
      className="sc-htpNat jMFgIO"
      id="EstratégiaValue"
      name="EstratégiaValue"
    >
      3x1
    </label>
  </div>
</div>
`;

exports[`Header when is enabled Rendering renders according to design 1`] = `
<div
  className="sc-ifAKCX LlTRb"
>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Ativo"
    >
      Ativo:
    </label>
    <select
      className="sc-bxivhb ldZUon"
      id="Ativo"
      name="Ativo"
      onChange={[Function]}
    >
      <option
        value="WDO"
      >
        WDO
      </option>
      <option
        value="WIN"
      >
        WIN
      </option>
    </select>
  </div>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Tipo"
    >
      Tipo:
    </label>
    <select
      className="sc-bxivhb ldZUon"
      id="Tipo"
      name="Tipo"
      onChange={[Function]}
    >
      <option
        value="Pregão"
      >
        Pregão
      </option>
      <option
        value="Abertura"
      >
        Abertura
      </option>
    </select>
  </div>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Sistema"
    >
      Sistema:
    </label>
    <select
      className="sc-bxivhb ldZUon"
      id="Sistema"
      name="Sistema"
      onChange={[Function]}
    >
      <option
        value={1}
      >
        name 1
      </option>
      <option
        value={2}
      >
        name 2
      </option>
    </select>
  </div>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Estratégia"
    >
      Estratégia:
    </label>
    <select
      className="sc-bxivhb ldZUon"
      id="Estratégia"
      name="Estratégia"
      onChange={[Function]}
    >
      <option
        value="Topo / Fundo"
      >
        Topo / Fundo
      </option>
      <option
        value="Fecha Fora / Fecha Dentro"
      >
        Fecha Fora / Fecha Dentro
      </option>
      <option
        value="3x1"
      >
        3x1
      </option>
      <option
        value="Fabolous 4"
      >
        Fabolous 4
      </option>
    </select>
  </div>
</div>
`;
